base/pointer.jl

```diff
@@ -99,8 +102,12 @@ Load a value of type `T` from the address of the `i`th element (1-indexed) start
 This is equivalent to the C expression `p[i-1]`.
 
 The `unsafe` prefix on this function indicates that no validation is performed on the
-pointer `p` to ensure that it is valid. Incorrect usage may segfault your program or return
-garbage answers, in the same manner as C.
+pointer `p` to ensure that it is valid. Like C, the programmer is responsible for ensuring
+that referenced memory is not freed or garbage collected while invoking this function.
+Incorrect usage may segfault your program or return garbage answers. Unlike C, dereferencing
+memory region allocated as different type may be valid provided that the types are compatible
+(e.g., `p` has the correct alignment, the stored bit pattern is known to be valid as `T`,
+`T` does not contain any GC-managed objects).
```

Comment:
```suggestion
memory region allocated as different type may be valid provided that the types are compatible.
```

None of these are particularly required.